### PR TITLE
Add sql top queries spec

### PR DESCRIPTION
**Pull request by jaredmoo**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jaredmoo** [comment]
Please don't merge until I get signoff from SQL DB query store team. Thx

**Comment by veronicagg** [comment]
@ravbhatnagar new spec being added, please take a look.

File: specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json
Comment (veronicagg) [inline specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json]: "type" missing for this parameter - causing CI to fail with syntax failure (not a valid swagger document) https://travis-ci.org/Azure/azure-rest-api-specs/jobs/257801073#L3217

**Review by veronicagg — CHANGES_REQUESTED** [review]
should the new spec be added to any of the packages in the "readme.md" file at azure-rest-api-specs/specification/sql/resource-manager/ ?

File: specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json]: Oops. Fixed.

**Comment by jaredmoo** [comment]
The spec will be added to "schema-2014-04" tag when #1474 is merged, and will be added to "package-2015-05-preview" concurrently with .net sdk code and tests are submitted for review.

File: specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json
Comment (veronicagg) [inline specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json]: looks like "required" value is in quotes, but should be a boolean - "required": true
CI still fails due to this, you can use the schema at https://raw.githubusercontent.com/Azure/autorest/master/schema/swagger-extensions.json to validate the file before submission.

**Review by veronicagg — CHANGES_REQUESTED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json]: @jaredmoo - Please add the id, name, type top level properties in the response. Id example below
id: "/subscriptions/{subid}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/topQueries/{queryNameOrId}"

File: specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json]: And then all these properties should move under the properties bag.

File: specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json]: IMO, you don't need a "statistics" type, a GET on a query returns the statistics information as well. So the API signature will be like - "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/topQueries/{queryId}

File: specification/sql/resource-manager/readme.md
Comment (veronicagg) [inline specification/sql/resource-manager/readme.md]: just wondering should 'schema-2014-04' be the new default tag?  - see line 31 in this file.

**Comment by veronicagg** [comment]
@jaredmoo thanks, added one more question inline. Waiting on review from @ravbhatnagar since it's a new file. 
Please let us know when you get signoff from SQL DB query store team

File: specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json]: A few Naïve questions to help my understanding - What categorizes a query into top queries? Is it the number of executions? Do you already have an API for listing all queries under a database? 
If yes, you can consider adding a property to response of /queries to indicate whether a query is a top query or not. And then to just list top queries, customers can use $filter on that property? 
All this is based on my limited knowledge of the scenario.

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]

**Comment by jaredmoo** [comment]
@veronicagg no, schema-2014-04 is what I'm using for conveniently generating RP schema and running --azure-validator until https://github.com/Azure/autorest/issues/2470 is fixed.

@ravbhatnagar this is all good feedback, but it's an old API that I can't change. Just adding missing swagger.

**Comment by ravbhatnagar** [comment]
@jaredmoo - ok :). One thing I would still like to point out - If a customer has defined a custom role where in NotActions he has explicitly added /servers/databases/topqueries/read, he will be able to still see all the top queries since you dont have ID in the response payload. Thus potentially breaking RBAC? Please ACK if you are fine with this.

**Comment by jaredmoo** [comment]
ACK - this API is a little weird by modern ARM standards, if/when we make a new api-version of this API it will be differently designed.

**Comment by veronicagg** [comment]
@jaredmoo ok, let me know when you get sign off from SQL DB query store team, so I can approve & merge. thanks!

File: specification/sql/resource-manager/Microsoft.Sql/2014-04-01/examples/QueriesList.json
Comment (sasapopo) [inline specification/sql/resource-manager/Microsoft.Sql/2014-04-01/examples/QueriesList.json]: Example is not correct. If numberOfTopQueries is 5 "queries" list below must have 5 items and it has only one.

File: specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json
Comment (sasapopo) [inline specification/sql/resource-manager/Microsoft.Sql/2014-04-01/queries.json]: DTU usage % corresponding to query categorizes it into top queries. We calculate DTU usage based on real resource usage recorded by Query Store.
We don't have api for listing all queries.

**Review by sasapopo — APPROVED** [review]

**Comment by veronicagg** [comment]
@jaredmoo do you want to update your example based on feedback from @sasapopo ?

**Review by veronicagg — APPROVED** [review]

**Pull request merged**
